Framework/DataHandling/src/AlignAndFocusPowderSlim.cpp

```diff
@@ -649,13 +658,64 @@ AlignAndFocusPowderSlim::determinePulseIndices(const API::MatrixWorkspace_sptr &
       throw std::invalid_argument("No valid pulse time indices found for filtering");
   }
 
-  // update the run TimeROI and remove log data outside the time ROI
-  wksp->mutableRun().setTimeROI(roi);
-  wksp->mutableRun().removeDataOutsideTimeROI();
-
   return pulse_indices;
 }
 
+/**
+ * @brief Determine the pulse indices for a given workspace, time ROI, and time splitter.
+ *
+ * @param wksp The workspace to get the pulse times from.
+ * @param roi The time ROI to use for filtering.
+ * @param timeSplitter The time splitter to use for determining target indices and additional time ROIs.
+ * @return std::vector<std::pair<int, PulseROI>> A vector of pairs, where each pair contains a target index and a
+ * PulseROI representing the pulse indices to include.
+ */
+std::vector<std::pair<int, PulseROI>>
+AlignAndFocusPowderSlim::determinePulseIndicesTargets(const API::MatrixWorkspace_sptr &wksp, const TimeROI &roi,
+                                                      const TimeSplitter &timeSplitter) {
+  // get pulse times from frequency log on workspace
+  const auto frequency_log = dynamic_cast<const TimeSeriesProperty<double> *>(wksp->run().getProperty("frequency"));
+  if (!frequency_log) {
+    throw std::runtime_error("Frequency log not found in workspace run");
+  }
+  const auto pulse_times =
+      std::make_unique<std::vector<Mantid::Types::Core::DateAndTime>>(frequency_log->timesAsVector());
+
+  std::vector<PulseROI> pulse_indices;
+  if (roi.useAll()) {
+    pulse_indices.emplace_back(0, std::numeric_limits<size_t>::max());
+  } else {
+    pulse_indices = roi.calculate_indices(*pulse_times);
+    if (pulse_indices.empty())
+      throw std::invalid_argument("No valid pulse time indices found for filtering");
+  }
+
+  const auto target_to_pulse_indices = timeSplitter.calculate_target_indices(*pulse_times);
+
+  // calculate intersection of target pulse indices with overall pulse indices, create new vector
```

Comment:
Is this to skip things that are being filtered out by other criteria? The comment doesn't quite make it clear enough for me.

Reply:
I updated some comments and changed some variable names to hopeful make this clearer.